stout/flags.cc

```diff
@@ -101,6 +188,34 @@ void Parser::Parse(int* argc, const char*** argv) {
 
     // Skip anything that doesn't look like a flag.
     if (arg.find("--") != 0) {
+      if (subcommands_.count(arg) > 0) {
+        std::cerr << "Encountered duplicate subcommand '"
+                  << arg << "'." << std::endl;
+        exit(1);
+      }
+      if (previous_subcommand_message_
+                  .value()
```

Comment:
But what if you can't call `.value()` because the optional is empty?

It seems like you always expect this to be set, so let's not use optional. Instead, let's just use a pointer and add `CHECK_NOTNULL` everywhere.